app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    results = {}
+
+    @wraps(func)
+    def inner(*args: Any, **kwargs: Any) -> Any:
+        key = (func.__name__, args, tuple(kwargs.items()))
```

Comment:
Why do you need name of function in your key? Each decorated function will have separate `results` dict

Reply:
I thought that if different functions with the same arguments were called, I wouldn't get "Getting from cache". But I can see that the code works without it, but I don't understand how... How does the code understand that these are two different functions, having only the arguments as the key and no information about the function name?